pkg/util/strategicpatch/patch.go

```diff
@@ -31,7 +31,352 @@ import (
 // lists should be merged or replaced.
 //
 // For more information, see the PATCH section of docs/api-conventions.md.
+
+// CreateStrategicMergePatch creates a patch that can be passed to StrategicMergePatch.
+// 'a' is original, 'b' is the modified document. Both are to be given as json encoded content.
+// The function will return a mergeable json document with differences from a to b.
+//
+// An error will be returned if either of the two documents is invalid.
+//
+// This method and several of its helpers were adapted from evanphx/json-patch.
+
+const specialKey = "$patch"
+
+var errBadJSONDoc = fmt.Errorf("Invalid JSON document")
+var errNoListOfLists = fmt.Errorf("Lists of lists are not supported")
+
+func CreateStrategicMergePatch(a, b []byte, dataStruct interface{}) ([]byte, error) {
```

Comment:
I would prefer the variable names original, modified vs a, b to keep consistent with StrategicMergePatch's original, patch, but that's just a personal preference and up to you if you want to change it.